clients/da-vinci-client/src/main/java/com/linkedin/davinci/helix/LeaderFollowerPartitionStateModel.java

```diff
@@ -91,14 +92,18 @@ public void onBecomeStandbyFromOffline(Message message, NotificationContext cont
       String storeName = Version.parseStoreFromKafkaTopicName(resourceName);
       int version = Version.parseVersionFromKafkaTopicName(resourceName);
       Store store = getStoreRepo().getStoreOrThrow(storeName);
-      boolean isRegularStoreCurrentVersion = store.getCurrentVersion() == version;
+      int currentVersion = store.getCurrentVersion();
+      boolean isCurrentVersion = currentVersion == version;
+      boolean isFutureVersionOnline =
+          currentVersion < version && store.getVersionStatus(version).equals(VersionStatus.ONLINE);
```

Comment:
Should we add another safeguard to check for deferred swap version config? because normally this should not happen.

Reply:
With deferred swap we might have a huge time difference from when the push job is completed and version is swapped, so this issue is more prominent. But this also can happen in normal scenarios but because the version swap is pretty quick, we don't see it. But its good to have this generic for all cases.